src/cool_seq_tool/sources/uta_database.py

```diff
@@ -271,68 +270,30 @@ def _transform_list(li: list) -> list[list[Any]]:
+    async def get_genomic_ac_genes(
         self,
         pos: int,
-        strand: Strand | None = None,
-        chromosome: int | None = None,
-        alt_ac: str | None = None,
-        gene: str | None = None,
-    ) -> tuple[GenesGenomicAcs | None, str | None]:
-        """Return genes and genomic accessions for a position on a chromosome or alt_ac
+        alt_ac: str,
+    ) -> tuple[set[str] | None, str | None]:
+        """Get genes given a genomic accession and position
 
         :param pos: Genomic position
-        :param strand: Strand
-        :param chromosome: Chromosome. Must give chromosome without a prefix
-            (i.e. ``1`` or ``X``). If not provided, must provide ``alt_ac``.
-            If ``alt_ac`` is also provided, ``alt_ac`` will be used.
-        :param alt_ac: Genomic accession (i.e. ``NC_000001.11``). If not provided,
-            must provide ``chromosome``. If ``chromosome`` is also provided, ``alt_ac``
-            will be used.
-        :param gene: Gene symbol
-        :return: Genes and genomic accessions and warnings if found
+        :param alt_ac: RefSeq
+        :return: Set of HGNC gene symbols associated to genomic accession and position
+            and warning
         """
-        alt_ac_cond = (
-            f"WHERE alt_ac = '{alt_ac}'"
-            if alt_ac
-            else f"WHERE alt_ac ~ '^NC_[0-9]+0{chromosome}.[0-9]+$'"
-        )
-        strand_cond = f"AND alt_strand = '{strand.value}'" if strand else ""
-        gene_cond = f"AND hgnc = '{gene}'" if gene else ""
-
         query = f"""
-            SELECT hgnc, alt_ac
+            SELECT DISTINCT(hgnc)
             FROM {self.schema}.tx_exon_aln_v
-            {alt_ac_cond}
+            WHERE alt_ac = '{alt_ac}'
             AND alt_aln_method = 'splign'
-            AND {pos} BETWEEN alt_start_i AND alt_end_i
-            {strand_cond}
-            {gene_cond};
+            AND {pos} BETWEEN alt_start_i AND alt_end_i;
             """  # noqa: S608
-
         results = await self.execute_query(query)
         if not results:
-            msg = (
-                f"Unable to find a result for chromosome "
-                f"{alt_ac or chromosome} where genomic coordinate {pos}"
-                f" is mapped between an exon's start and end coordinates"
-            )
-            if strand:
-                msg += (
-                    f" on the "
-                    f"{'positive' if strand == Strand.POSITIVE else 'negative'} strand"
-                )
-            if gene:
-                msg += f" and on gene {gene}"
-            return None, msg
+            return None, f"No gene(s) found given {alt_ac} on position {pos}"
 
-        results = self._transform_list(results)
-        genes = set()
-        alt_acs = set()
-        for r in results:
-            genes.add(r[0])
-            alt_acs.add(r[1])
-        return GenesGenomicAcs(genes=genes, alt_acs=alt_acs), None
+        return {r["hgnc"] for r in results}, None
```

Comment:
do we need to define it as a set since we're already selecting for DISTINCT gene names?

Reply:
Nah I can change to list